src/sasl-sha256.js

```diff
@@ -0,0 +1,33 @@
+import MD5 from './md5';
+import SASLMechanism from './sasl.js';
+import utils from './utils';
+import scramResponse from './scram.js'
+
+export default class SASLSHA256 extends SASLMechanism {
+
+    /** PrivateConstructor: SASLSHA256
+     *  SASL SCRAM SHA 256 authentication.
+     */
+    constructor (mechname='SCRAM-SHA-256', isClientFirst=true, priority=70) {
+        super(mechname, isClientFirst, priority);
+    }
+
+    test (connection) { // eslint-disable-line class-methods-use-this
+        return connection.authcid !== null;
+    }
+
+    async onChallenge (connection, challenge) { // eslint-disable-line class-methods-use-this
+        return await scramResponse(connection, challenge, "SHA-256", 256);
+    }
+
+    clientChallenge (connection, test_cnonce) {  // eslint-disable-line class-methods-use-this
```

Comment:
Looks like this `clientChallenge` method is the same for all three SCRAM-SHA mechanisms, so it would be preferable to have it defined once in `scram.js`, like you did with `scramResponse`.

Reply:
Unfortunately, because the SHA3 interface is not yet available in webcrypto, the sha3 scram mechanisms (which I haven't tested yet at all) do not yet fit the same pattern as the sha2 variants. I can definitely factor this part out, though. I'll take a look at it.

Reply:
> Unfortunately, because the SHA3 interface is not yet available in webcrypto, the sha3 scram mechanisms (which I haven't tested yet at all) do not yet fit the same pattern as the sha2 variants. I can definitely factor this part out, though. I'll take a look at it.

scratch that, it is the same. I thought it was different for some reason.